### DISCUSSION TITLE
Does KiBot control DRC configuration options?

### DISCUSSION THREAD

**Discussion by cdwilson** [INTI-CMNB/KiBot#297] [Q&A]
KiCad 6 has some configuration options for DRC which can be enabled/disabled via the GUI:

<img width="691" alt="image" src="https://user-images.githubusercontent.com/14631/190516016-d41865af-0a0d-4602-9b86-ba8b5d1d5413.png">

Does KiBot set these options or does it just used whatever is saved in the KiCad project?  If it sets these options, what are the defaults?  (I want to make sure that these options are set when KiBot runs)

**Comment by set-soft** [comment]
Hi @cdwilson !
KiAuto always enables the options on the left side (refill_zones and test_all_track_errors).
The option on the right (test_footprints) is not enabled because it isn't available. This option is only available when (both):

1. You run the KiCad GUI
2. You start pcbnew from the kicad frontend, starting pcbnew alone doesn't even show this option. 

The test_footprints option seems to need to run eeschema. I'm not familiarized with what exactly is done. I guess it checks that the `footprint` field of the symbols matches the actual footprint. But I don't know if this is all. This check could be implemented in KiBot.

**Reply by cdwilson** [reply]
> I guess it checks that the footprint field of the symbols matches the actual footprint. But I don't know if this is all.

I think it might also do a netlist comparison between the schematic and the PCB.

The official docs aren't super helpful:

> **Test for parity between PCB and schematic:** when enabled, the design rule checker will test for differences between the schematic and PCB in addition to testing the PCB design rules. This option has no effect when running the PCB editor in standalone mode.

However, I tried making some changes to the schematic and then I ran the DRC with the "Test for parity between PCB and schematic" checkbox enabled, and KiCad reported the differences in the netlist:

<img width="843" alt="image" src="https://user-images.githubusercontent.com/14631/190794613-43122e4b-c28c-470a-b7b4-f80c0c79eb7a.png">

**Reply by set-soft** [reply]
@cdwilson 
Ok, I added an option to `update_xml` it does some similar work.
It loads the generated XML and compares it with the PCB data. 
The checks are to see if the components matches, note that the schematic must be fully annotated to get meaningful results.
It also checks each netlist to see if the name matches and if they connect the same pads.
I guess it will need various adjusts, but I got a good result using a very simple project with various errors introduced on purpose.
